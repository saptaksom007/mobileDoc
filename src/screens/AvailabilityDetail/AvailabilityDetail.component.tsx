import React, { PureComponent } from 'react'
import { connect } from 'react-redux'

import { ScrollView } from 'react-native'
import { compose } from 'recompose'
import i18n from 'ex-react-native-i18n'
import { Card, ListItem } from 'react-native-elements'
import withSubtitle from 'screens/enhancers/withSubtitle'
import { Bold, Medium } from 'components/CustomText/CustomText.component'
import { fontFamily } from 'components/CustomText/CustomText.styles'
import MessageMyDoctor from 'components/MessageMyDoctor/MessageMyDoctor.component'
import { isInteger } from 'utilities/number'
import PhoneNumber from 'components/PhoneNumber/PhoneNumber.component'

import styles from './AvailabilityDetail.styles'

interface Props {
  navigation?: {
    state: {
      params: {
        clinicName: string
        clinicStreet: string
        clinicLocation: string
        clinicContact: string
        clinicMessage: string
        doctorMessage: string
        physicianRef: string
      }
    }
  }
}

export class AvailabilityDetail extends PureComponent<Props> {
  renderClinic() {
    const {
      clinicName,
      clinicStreet,
      clinicLocation,
      clinicContact
    } = this.props.navigation!.state.params
    const hasStreet = clinicStreet && clinicStreet !== ''
    const hasCity = clinicLocation && clinicLocation !== ''
    return (
      <Medium>
        <Bold>{i18n.t('availabilitydetail.card.clinic')}:</Bold> {clinicName}
        {'\n'}
        {(hasCity || hasStreet) && (
          <Bold>{i18n.t('availabilitydetail.card.address')}: </Bold>
        )}
        {hasStreet && (
          <Medium>
            {clinicStreet}
            {', '}
          </Medium>
        )}
        {hasCity && <Medium>{clinicLocation}</Medium>}
        {clinicContact && (
          <Medium>
            {'\n'}
            <Bold>{i18n.t('availabilitydetail.card.phone')}:</Bold>{' '}
            <PhoneNumber value={clinicContact} />
          </Medium>
        )}
      </Medium>
    )
  }

  renderClinicAvailability() {
    const { clinicMessage } = this.props.navigation!.state.params
    return (
      <Medium style={styles.centered}>
        <Bold>{i18n.t('availabilitydetail.card.clinicAvailability')}</Bold>
        {'\n'}
        <Medium>
          {isInteger(clinicMessage)
            ? i18n.t('availabilitydetail.card.clinicMessage', {
              hours: clinicMessage
            })
            : clinicMessage}
        </Medium>
      </Medium>
    )
  }

  renderDoctorAvailability() {
    const { doctorMessage } = this.props.navigation!.state.params
    return (
      <Medium style={styles.centered}>
        <Bold>{i18n.t('availabilitydetail.card.doctorAvailability')}</Bold>
        {'\n'}
        <Medium>{doctorMessage}</Medium>
      </Medium>
    )
  }

  render() {
    const {
      clinicName,
      clinicMessage,
      doctorMessage,
      physicianRef
    } = this.props.navigation!.state.params

    const clinic =
      !clinicMessage || clinicMessage === '' ? null : (
        <ListItem title={this.renderClinicAvailability()} hideChevron />
      )

    const doctor =
      !doctorMessage || doctorMessage === '' ? null : (
        <ListItem
          title={this.renderDoctorAvailability()}
          hideChevron
          containerStyle={styles.cardLastItemNoBorder}
        />
      )
    return (
      <ScrollView>
        <Card
          title={`${i18n.t('availabilitydetail.card.title')} - ${clinicName}`}
          containerStyle={styles.cardContainerStyle}
          titleStyle={styles.cardTitle}
          fontFamily={fontFamily}
        >
          <ListItem title={this.renderClinic()} hideChevron />
          {clinic}
          {doctor}
        </Card>
        <MessageMyDoctor
          otherFilter={(conversation: any) =>
            conversation.subscriptions.find(
              (sub: any) => sub.userRef === physicianRef
            )
          }
        />
      </ScrollView>
    )
  }
}

function select(): any {
  return {}
}

export default compose<Props, any>(
  connect(select),
  withSubtitle('availabilitydetail.subtitle')
)(AvailabilityDetail)
