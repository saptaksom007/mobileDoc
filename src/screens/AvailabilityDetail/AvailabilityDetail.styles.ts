import { StyleSheet } from 'react-native'
import { Color } from 'constants/Color'

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    justifyContent: 'center',
    alignItems: 'center'
  },
  centered: {
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center'
  },
  cardContainerStyle: {
    margin: 20
  },
  cardTitle: {
    color: Color.lightText
  },
  cardLastItemNoBorder: {
    borderBottomWidth: 0
  }
})
