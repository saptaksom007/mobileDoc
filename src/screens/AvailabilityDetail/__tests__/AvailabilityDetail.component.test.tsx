import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { AvailabilityDetail } from '../AvailabilityDetail.component'

jest.mock(
  'components/MessageMyDoctor/MessageMyDoctor.component',
  () => 'MessageMyDoctor'
)

it('<AvailabilityDetail/> render correctly', () => {
  const tree = renderer
    .create(
      <AvailabilityDetail
        navigation={{
          state: {
            params: {
              clinicName: 'clinicName',
              clinicStreet: 'clinicStreet',
              clinicLocation: 'clinicLocation',
              clinicContact: 'clinicContact',
              clinicMessage: 'clinicMessage',
              doctorMessage: 'doctorMessage',
              physicianRef: 'physicianRef'
            }
          }
        }}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
