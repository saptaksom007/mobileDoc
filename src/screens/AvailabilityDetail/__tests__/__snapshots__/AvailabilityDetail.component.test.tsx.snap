// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvailabilityDetail/> render correctly 1`] = `
<RCTScrollView>
  <View>
    <MessageMyDoctor
      otherFilter={[Function]}
    />
  </View>
</RCTScrollView>
`;
